validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/DepositProvider.java

```diff
@@ -80,14 +80,22 @@ public synchronized void onDepositsFromBlock(DepositsFromBlockEvent event) {
   }
 
   @Override
-  public synchronized void onNewFinalizedCheckpoint(final Checkpoint checkpoint) {
-    BeaconState finalizedState =
-        recentChainData
-            .getBlockState(checkpoint.getRoot())
-            .orElseThrow(
-                () -> new IllegalArgumentException("Finalized Checkpoint state can not be found."));
-
-    depositNavigableMap.headMap(finalizedState.getEth1_deposit_index(), false).clear();
+  public void onNewFinalizedCheckpoint(final Checkpoint checkpoint) {
+    recentChainData
+        .retrieveBlockState(checkpoint.getRoot())
+        .thenAccept(
+            finalizedState -> {
+              if (finalizedState.isEmpty()) {
+                throw new IllegalStateException("Finalized Checkpoint state cannot be found.");
```

Comment:
Probably should just log an error here and exit rather than throwing.  It should never happen so probably doesn't matter but if the checkpoint state isn't available it's not actually that big a deal - we just won't prune deposits so don't want to be too noisy about it.